xla/service/gpu/matmul_utils.h

```diff
@@ -43,6 +43,31 @@ limitations under the License.
 namespace xla {
 namespace gpu {
 
+
+inline void GetXlaPrecisionConfigNumericFlags(const mlir::ArrayAttr& precision_config,
+  int& grad_flags, int64_t& compute_precision)
+{
+    compute_precision = 0;
+    grad_flags = 0;
+    int grad_count=0;
+    for (auto attr : precision_config) {
+      int64_t value = static_cast<int64_t>(
+          attr.template cast<mlir::mhlo::PrecisionAttr>().getValue());
+      if (value > compute_precision && value < PrecisionConfig::ACTIVATION) {
+        compute_precision = value;
+      }
+      if(value>=PrecisionConfig::ACTIVATION && value<=PrecisionConfig::GRADIENT)
+      {
+        if(grad_count<2 && value == PrecisionConfig::GRADIENT)
+          grad_flags |= 1 << grad_count;
+        grad_count++;
+      }
+    }
+    if(grad_count==2)
+      grad_flags |= 256;
```

Comment:
What is this '256' constant? Can it be pulled out into a constant?

Reply:
It is an indication that the field has been initialized, I have it declared as NumericOptions::GF_Initialized (not sure if that declaration is visible at this point in the header). I'll try to get rid of 256.